ui/vty-lib/src/Ariadne/UI/Vty/Widget.hs

```diff
@@ -393,21 +414,27 @@ withWidgetState action = do
 -- Widget rendering
 ----------------------------------------------------------------------------
 
-drawWidget :: WidgetName -> p -> Widget p -> B.Widget WidgetName
+drawWidget :: WidgetName -> p -> Widget p -> WidgetDrawing
 drawWidget focus parent (Widget widget@WidgetInfo{..}) =
-    visible $ runReader (runReaderT (widgetDrawWithFocus focus widgetState) widget) parent
+    map visible $ runReader (runReaderT (widgetDrawWithFocus focus widgetState) widget) parent
   where
     visible w
       | focus == widgetName
       , not widgetIgnoreVisibility = B.visible w
       | otherwise = w
 
-drawWidgetChild :: WidgetName -> WidgetInfo s p -> WidgetNamePart -> B.Widget WidgetName
+drawWidgetChild :: WidgetName -> WidgetInfo s p -> WidgetNamePart -> WidgetDrawing
 drawWidgetChild focus widget@WidgetInfo{..} name =
-    maybe B.emptyWidget (drawWidget focus widget . withIgnoreVisibility) $ Map.lookup name widgetChildren
+    singleDrawing . maybe B.emptyWidget (last . drawWidget focus widget . withIgnoreVisibility) $ Map.lookup name widgetChildren
   where
     withIgnoreVisibility (Widget child) = Widget child{ widgetIgnoreVisibility = widgetIgnoreVisibility }
 
+singleDrawing :: (B.Widget WidgetName) -> WidgetDrawing
+singleDrawing = NE.fromList . (: [])
```

Comment:
Isn't that just `(:| [])`? Also, redundant bracket in type signature.

Reply:
Yes, please avoid using `NE.fromList` wherever possible, because it's a partial function.